test/e2e/framework/helpers/k8s/k8s_utils.go

```diff
@@ -1241,12 +1242,12 @@ func (k *KubeCtl) WaitForPlaceholders(namespace string, podPrefix string, numPod
 
 func (k *KubeCtl) ListPlaceholders(namespace string, podPrefix string) ([]v1.Pod, error) {
 	pods := make([]v1.Pod, 0)
-	podList, lstErr := k.ListPods(namespace, "placeholder=true")
+	podList, lstErr := k.ListPods(namespace, "")
 	if lstErr != nil {
 		return pods, lstErr
 	}
 	for _, pod := range podList.Items {
-		if strings.HasPrefix(pod.Name, podPrefix) {
+		if strings.HasPrefix(pod.Name, podPrefix) && pod.Annotations[constants.AnnotationPlaceholderFlag] == constants.True {
```

Comment:
Since we cannot list placeholder pods directly using LabelSelector anymore, I retrieve all pods and then check placholder pod by pod.

Reply:
What will happen if users rolling-update yunikorn with existent running pods? Those pods having deprecated label don't have the new annotation.

Reply:
@chia7712 
In 1.6.0, the function GetPlaceholderFlagFromPodSpec() fetch placholder from pod in below order:

1. internal.yunikorn.apache.org/placeholder
2. yunikorn.apache.org/placeholder  (Deprecated Annotation)
3. placeholder (Deprecated Label)

The existing placeholder pod is still considered as placeholder pod in 1.6.0.
I'll remove deprecated placeholder in 1.7.0.